mod/blog/elgg-plugin.php

```diff
@@ -14,6 +14,57 @@
 		'blog/auto_save_revision' => [],
 		'blog/delete' => [],
 	],
+	'routes' => [
+		'blog_owner' => [
+			'path' => '/blog/owner/{username?}',
+			'resource' => 'blog/owner',
+		],
+		'blog_friends' => [
+			'path' => '/blog/friends/{username?}',
+			'resource' => 'blog/friends',
+		],
+		'blog_archive' => [
+			'path' => '/blog/archive/{username?}/{lower?}/{upper?}',
+			'resource' => 'blog/archive',
+			'requirements' => [
+				'lower' => '\d+',
+				'upper' => '\d+',
+			],
+		],
+		'blog_view' => [
+			'path' => '/blog/view/{guid}/{title?}',
+			'resource' => 'blog/view',
+		],
+		'blog_add' => [
+			'path' => '/blog/add/{guid?}',
+			'resource' => 'blog/add',
+		],
+		'blog_edit' => [
```

Comment:
For `edit` we need some system-wide approach, so we can have `/edit` as the primary route, which then renders a resource that has a predictable name `/resources/edit/<type>/<subtype>`